internal/photoprism/mediafile.go

```diff
@@ -314,6 +317,86 @@ func (m *MediaFile) EditedName() string {
 	return ""
 }
 
+func (m *MediaFile) ExtractEmbeddedVideo() (string, error) {
+	if m == nil {
+		return "", fmt.Errorf(
+			"mediafile: file is nil - you may have found a" +
+				" bug",
+		)
+	}
+
+	// Abort if the source media file does not exist.
+	if !m.Exists() {
+		return "", fmt.Errorf(
+			"mediafile: %s not found", clean.Log(m.RootRelName()),
+		)
+	} else if m.Empty() {
+		return "", fmt.Errorf(
+			"mediafile: %s is empty", clean.Log(m.RootRelName()),
+		)
+	}
+
+	// get the embedded video field name from the file metadata
+	if metaData := m.MetaData(); metaData.Error == nil && metaData.
+		EmbeddedVideo != "" {
+		path := filepath.Join(
+			Config().SidecarPath(), m.RootRelPath(), "%f",
+		)
+		cmd := exec.Command(
+			Config().ExifToolBin(), fmt.Sprintf("-%s", metaData.EmbeddedVideo),
+			"-b", "-w",
+			path, m.FileName(),
+		)
+
+		var out bytes.Buffer
+		var stderr bytes.Buffer
+		cmd.Stdout = &out
+		cmd.Stderr = &stderr
+		cmd.Env = []string{fmt.Sprintf("HOME=%s", Config().CmdCachePath())}
```

Comment:
`HOME=Config().CmdCachePath()` is a hack we originally added for Darktable because it requires the home directory to be writable even if no (or only temporary) files are created there. To get the configured tmp file path, you can use `Config().TempPath()`.

Reply:
Note that it doesn't hurt to keep this, even though it may not be required for Exiftool. It MAY be needed because Docker containers can run with UIDs for which no home (or actual user account) exists in the container.